### ISSUE TITLE
test(model): independently verify V4.1 latent norms through full forward

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#13322]
# test(model): independently verify V4.1 latent norms through full forward

```routing
repo: fak
lane: model
paths: ["internal/model/v41_forward_test.go","internal/model/v41_latent_norm_parity_test.go"]
expected_steps: 3
```

## Working spine
Expose V4.1 integration failures before another full-model hardware run, using existing runtime and test surfaces.

## Current state
The current latent-norm test's oracle and comparison both call production rmsnormCfg and matRows (v41_attn_latent_norm_test.go:44-47,90-93). Its full-path arm usefully detects non-finite overflow. The separate v41ReducedModel/v41OracleForward fixture omits latent norm tensors and their numerical effect. These are distinct gaps; this leaf preserves the existing overflow regression.

## Why this is next
The native mission has reached a token but continues through expensive sequential bring-up. This leaf removes one class of uncertainty in a bounded software check while the current performance work proceeds.

## Parent context
anthony-chaudhary/fak#12640; engineering support for anthony-chaudhary/fak#13294 and MISSION-halo-v41-native. Existing deepseek-v41-flash and public issue missions can dispatch this single-package leaf.

## Core through-line
Extend the independent scalar full-forward oracle to apply optional Q-latent and KV-latent norm tensors with the pinned V4.1 order. Add an opt-in reduced fixture carrying non-unit, nonuniform gains and nontrivial projections, leaving existing goldens unchanged. Compare actual Model.Forward logits for that fixture with independent scalar math; expected calculations must not call rmsnormCfg, matRows, matVecV41Weight or other production helpers.

## Gold-plating boundary
Test-only; no production math edits, regenerated baseline goldens, general oracle rewrite, hardware qualification, V4-0731 equations, or performance claims. Keep the existing default fixture byte-for-byte equivalent.

## Worker packet
DS4.1 or Qwen3.8 capability level; S1. Own only the paths below. Three steps: establish the specified failing/missing contract with an independent fixture; implement only the bounded change; run the named and owning-package gates and land with explicit paths. No open-ended study or architecture choice remains. If a source symbol moved, locate its equivalent; do not silently widen the contract.

## Work unit
leaf

## Assumptions
- The pinned V4.1 architecture and existing production/test seams remain the authority; re-read changed symbols before editing.

## Confusion risks
- A passing synthetic witness proves the declared software envelope only, not physical performance or complete checkpoint equivalence.

## Coordination
- Acquire the listed public lane and preserve peer work. Serialize shared model-lane edits; use the dependencies section for start holds.

## Trigger
Operator-requested architecture planning after repeated sequential native bring-up failures; source-backed gaps documented in Current state.

## Batch policy
One issue per unique fak-native-shiftleft-key; search open and closed tickets before filing or expanding. Update the existing leaf on rerun.
## Work scale
S1

## Model tier
Required: T1
Optimal: T1

## Dependencies and scope fences
- Coordinates with: anthony-chaudhary/fak#13303 (window semantics remain outside this fixture).
- Start blocked by: none.

## Reference and reuse decision
ADAPT the exact [V4.1 Attention reference](https://huggingface.co/deepseek-ai/DeepSeek-V4.1-Flash/blob/dba1be0a40aa45a94ad051997016db3960a90277/inference/model.py#L790) (MIT): q_norm(wq_a(x)) before wq_b and kv_norm(wkv(x)) before rotary. Pin dba1be0a40aa45a94ad051997016db3960a90277, seed/config and calculation method in the test. This extends closed #12908/#13151; it does not reopen their completed scope or duplicate V4-0731 #12641.

## Reproduction witness
Source-backed coverage gap: inspect the functions and current tests named above at public revision 3ce59641b971a31b6bf19838290e64c0bf5d1033, then add the named regression. A missing test is not a passing baseline. Capture its RED result or explicit missing-coverage finding before changes; do not claim an unexecuted production failure.

## Done condition / witness
Witness: `go test ./internal/model -run '^TestV41ForwardParityLatentNorm$' -count=1 -v`
The test is a required deliverable, not asserted to exist today. Confirm the exact test appears in `go test ./internal/model -list '^TestV41ForwardParityLatentNorm$'`; empty matches and unexpected skips fail acceptance. Then run `go test ./internal/model -count=1` and required boundary gates after the last edit, in the provisioned public candidate worktree.

## Verifiable Witness
```
go test ./internal/model -run '^TestV41ForwardParityLatentNorm$' -count=1 -v
```

## Definition of done
- [ ] Actual Model.Forward agrees with independent reference math for nonuniform Q and KV gains on at least two prompts and a two-layer fixture using a fixed stated tolerance.
- [ ] Negative controls inside independent expected math omit a norm, move Q norm after wq_b, or substitute unit gains and must move expected logits beyond tolerance; do not remove required production tensors, which would test admission instead.
- [ ] Existing TestV41Parity and latent-overflow regressions still pass; new focused and owning-package tests run and pass after the final edit.

## Acceptance gate
Every checklist item is required; focused tests plus owning-package tests and independent review must pass. This is [SW-VERIFIED] coverage only. Physical throughput and coherent two-prompt qualification remain open with the existing mission owner.

## Blast radius and fallback
One package: internal/model. Existing behavior remains the fallback until this leaf passes. Preserve unrelated work, fail closed on unsupported shapes, and retain the prior physical mission gate. No new hardware run is a prerequisite.

## Closure binding
DCO sign-off, explicit paths, issue reference, and (fak model) commit subject suffix. Follow existing provenance and lease rules. Land via Fak Flow and push; no agent-created PR. Closure covers this leaf only.

## Likely files
- internal/model/v41_forward_test.go
- internal/model/v41_latent_norm_parity_test.go

## Expected steps
3

## Value
- Centrality: Core
- P1 Context: advanced - turns a known integration gap into a reusable local check
- P2 Net value: advanced - avoids repeated full-model diagnosis without relaxing correctness
- P3 Adaptation: preserved - reuses existing contracts and keeps runtime policy stable
- P4 Operations: advanced - bounded inputs and exact failure evidence support ordinary workers

## Work estimate
Estimate: 3 points

## Overall completion contribution
Contribution: 3/30 points

## Completion standard
development

## Qualification target
Pinned V4.1 text architecture; scoped synthetic or metadata-only software gate on a CPU development host.

## Planning baseline
Source inspection and planning only at filing; implementation and its tests are not yet witnessed.

**Comment by anthony-chaudhary** [comment]
Closed by landed commit `234a3b0293711064f0ad0ef58d25eec37a728000` (trunk, pushed to `origin/main`).

**Deliverable:** `internal/model/v41_latent_norm_parity_test.go` (+502) — an independent scalar full-forward oracle that transcribes the FULL V4.1 dataflow (flattened four-stream mHC projection, persistent residual streams, causal sink contraction, router + routed/shared experts, post-mix write-back) and interposes the Q/KV latent norms at the pinned reference order, plus an opt-in fixture with nonuniform non-unit gains.

**Why it is a real witness (not a tautology):**
- Oracle reuses none of the production machinery under test (`rmsnormCfg`, `matRows`, `v41ProjMatRows`, `v41MHCProjectFull`, `V41SparseAttentionSink`, `v41RopeTableForLayer`); it reads the fixture's raw f32 bytes and uses only weight-free scalar primitives.
- RED-on-regression verified: temporarily removing `qLat = rmsnormCfg(qLat, qNorm, eps, cfg)` from `v41_forward.go` made the test fail (|delta| = 8.03e-2 >> tol 1e-4); restored production passes.
- Negative controls inside the oracle (omit Q norm, omit KV norm, apply Q norm after wq_b, substitute unit gains) each move expected logits beyond `cpuOracleTol` (1e-4).

**Witness (on landed trunk):**
```
go test ./internal/model -run '^TestV41ForwardParityLatentNorm$' -count=1 -v  => PASS (both subtests)
go test ./internal/model -count=1                                              => ok (234.1s, no regressions)
go vet ./internal/model                                                        => exit 0
gofmt -l internal/model/v41_latent_norm_parity_test.go                         => clean
fak-boundary check (private 5-gate, worktree)                                  => PASS (0 violations, 5/5 gates)
```

Two sub-oracle gaps found and closed along the way (documented in the test): the published full config carries YaRN rope scaling (`rope_type: yarn`) and per-layer theta, which `v41RopeTableForLayer` folds in; and the full path accumulates the attention dot in float32, both of which a naive float64/bare-theta oracle would silently miss. Default reduced fixture and unit-gain full fixture are byte-for-byte unchanged.

Coordinates with #13294 (production ownership retains) and #13303. This is [SW-VERIFIED] coverage only; no physical throughput claim. Parent epic #12640 stays OPEN.
